packages/auth/src/platform_browser/popup_redirect.test.ts

```diff
@@ -177,6 +222,66 @@ describe('platform_browser/popup_redirect', () => {
       );
     });
 
+    it('includes the App Check token in the url fragment if present', async () => {
+      sinon
+        .stub(FAKE_APP_CHECK_CONTROLLER, 'getToken')
+        .returns(Promise.resolve({ token: 'fake-token' }));
+
+      // This promise will never resolve on purpose
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      resolver._openRedirect(auth, provider, event);
```

Comment:
I haven't tried this, but looks like you can await on _openRedirect.. In the signInWithRedirect docs, we do describe this in the example snippet:

https://firebase.google.com/docs/reference/js/auth.md#signinwithredirect

We can atleast wrap it in a then/catch as we do in the demo app - https://github.com/firebase/firebase-js-sdk/blob/825e648b81ca63c7bc64f8700f7a46eb320b2106/packages/auth/demo/src/index.js#L1347

Do you mind trying one of those and removing the 100ms await below? 
We can clean up the other test cases in this file in a later PR, it would be great if you can try it out for the new cases. Thanks!

Reply:
Yeah, so the issue here is that `_openRedirect()` returns `Promise<never>`, which never resolves but is an appropriate type for the method since it indeed will never return to the caller. I tried adding the `await` before along with some other things to make it work (e.g. casting the `Promise<never>` to `Promise<void>`) but to no avail. The timeout is a workaround to this

Reply:
Thanks for trying it. I also noticed now that the then/catch block I linked above (https://github.com/firebase/firebase-js-sdk/blob/825e648b81ca63c7bc64f8700f7a46eb320b2106/packages/auth/demo/src/index.js#L1347) is in case of popup only, where the new domain is opened in a popup rather than replacing the sdk/app.

So the workaround makes sense. We should cleanup the docs to remove the `await` keyword. Will do that.